whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticIndexService.java

```diff
@@ -48,23 +48,33 @@ public class ElasticIndexService {
 
     private static final String SERIAL_DOC_ID = "1";
 
-    private final RestHighLevelClient client;
-    private final String whoisIndex;
+    private RestHighLevelClient client;
+    private final String whoisAliasIndex;
     private final String metadataIndex;
 
     @Autowired
-    public ElasticIndexService(@Value("#{'${elastic.host:localhost:9200}'.split(',')}") final List<String> elasticHosts,
-                               @Value("${elastic.whois.index:whois}") final String whoisIndexName,
+    public ElasticIndexService(@Value("#{'${elastic.host:}'.split(',')}") final List<String> elasticHosts,
+                               @Value("${elastic.whois.index:whois}") final String whoisAliasName,
                                @Value("${elastic.commit.index:metadata}") final String whoisMetadataIndexName) {
-        this.whoisIndex = whoisIndexName;
+        this.whoisAliasIndex = whoisAliasName;
         this.metadataIndex = whoisMetadataIndexName;
-        final RestClientBuilder clientBuilder = RestClient.builder(elasticHosts.stream().map( (host) -> HttpHost.create(host)).toArray(HttpHost[]::new));
-        client = new RestHighLevelClient(clientBuilder);
+        getEsClient(elasticHosts);
+    }
+
+    private void getEsClient(final List<String> elasticHosts) {
+        try {
+            final RestClientBuilder clientBuilder = RestClient.builder(elasticHosts.stream().map((host) -> HttpHost.create(host)).toArray(HttpHost[]::new));
+            client = new RestHighLevelClient(clientBuilder);
+        } catch (Exception e) {
+            LOGGER.warn("Failed to tsrat the ES client {}", e.getMessage());
+        }
     }
 
     @PreDestroy
     public void preDestroy() throws IOException {
-        client.close();
+        if (isElasticRunning()) {
+            client.close();
```

Comment:
Check if client is null (uninitialised)

Reply:
Added a null check as well . So now it checks if client is not null and also it is reachable